### ISSUE TITLE
ERROR: findDependencyInfo: "ldd: /usr/lib/../lib/..: not regular file\n"

### ISSUE THREAD

**Issue by kanryu** [probonopd/linuxdeployqt#225]
Hi,

I add '-ldl' into LDFLAGS of application exectable, and it occars the error of the title.

what happened?

https://gist.github.com/kanryu/b00f5745203bcb0408460de21e3c54fc

**Comment by TheAssassin** [comment]
linuxdeployqt calls patchelf to read the rpath of a directory. It does this to determine whether to run `strip` to optimize your binary.

I am not sure where it got that path from, but I assume it's not the `-ldl` change which causes these issues.

You should provide some more information, though. What project are you working on? How did you call (q|c)make, linuxdeployqt, etc.?

**Comment by kanryu** [comment]
the log is for building QuickViewer https://github.com/kanryu/quickviewer

linuxdeployqt is run by qmake.

**Comment by probonopd** [comment]
For some weird reason, linuxdeployqt thinks that `usr/` is a binary and not a directory.

```
Log: checking whether "/home/kanryu/work/qvproject/QuickViewer-1.0.9-x64.AppDir/usr" has an rpath set
```

This is clearly wrong. We need to find out where this is coming from.

**Comment by probonopd** [comment]
> linuxdeployqt is run by qmake.

Just for fun, what happens if you do everything 1:1 as in this [example](https://github.com/probonopd/linuxdeployqt#using-linuxdeployqt-with-travis-ci)?

**Comment by TheAssassin** [comment]
@probonopd looks like there should be a check whether a file is really a file or a directory.

**Comment by kanryu** [comment]
build log is here.

https://gist.github.com/kanryu/b78b4352810fe60e81b337c069150632

**Comment by probonopd** [comment]
> @probonopd looks like there should be a check whether a file is really a file or a directory.

While this certainly wouldn't hurt, I'm curious why a directory is ending up there in the first place...

**Comment by probonopd** [comment]
> build log is here.

Can you please run with `-verbose=3`? Thanks.

**Comment by kanryu** [comment]
-verbose=3 log is here.

https://gist.github.com/kanryu/c643d018f43c657bb39ded0dc3661f97

**Comment by probonopd** [comment]
First occurrence of the wrong string is here:
https://gist.github.com/kanryu/c643d018f43c657bb39ded0dc3661f97#file-build-verbose3-log-L700

Clearly a `linuxdeployqt` bug. This should not happen.

So far I don't know yet where this is coming from. Would need to dig into this but am currently lacking the time. Anyone's guess is as good as mine...

**Comment by probonopd** [comment]
Which OS and version are you building on, and where is the Qt coming from?

**Comment by kanryu** [comment]
It is the official Qt SDK 5.9.1, and kubuntu-14.04.5-desktop

**Comment by TheAssassin** [comment]
Please define "official".

**Comment by kanryu** [comment]
What do you mean 'official'?

I can't find any official such as option of qmake and linuxdepoyqt.

**Comment by TheAssassin** [comment]
> It is the official Qt SDK 5.9.1 community, and kubuntu-14.04.5-desktop

I was referring to the "official" in there. Do you mean the pre-built upstream binaries, or the ones your system provides, ...?

**Comment by kanryu** [comment]
Qt SDK is not an ubuntu apt package.

**Comment by kanryu** [comment]
[QuickViewer-1.0.9-x64.AppDir.tar.gz](https://github.com/probonopd/linuxdeployqt/files/1676989/QuickViewer-1.0.9-x64.AppDir.tar.gz)

**Comment by TheAssassin** [comment]
That's not really a reply, but I guess you intended to imply that you use the upstream Qt.

And of course there's a perfectly usable Qt SDK in the Ubuntu repositories:

```
sudo apt-get install qt5-default qt5-qmake

**Comment by probonopd** [comment]
@TheAssassin he said 

> Qt SDK from https://www.qt.io/download-open-source/

@kanryu please try the exact OS and Qt combination from the project's `README.md` Travis CI example.

**Comment by TheAssassin** [comment]
@probonopd as you might see, that comment has been edited, that information was missing in the original version.

**Comment by kanryu** [comment]
> @kanryu please try the exact OS and Qt combination from the project's README.md Travis CI example.
Do you want me to use the Qt SDK provided by OS?

Since my build environment is not for using the application, there is no point in using the Qt provided by the build environment.

In the first place I have released many versions of QuickViewer in this environment, and it is no longer the first time I succeed.

**Comment by probonopd** [comment]
> Since my build environment is not for using the application, there is no point in using the Qt provided by the build environment.

With `linuxdeployqt`, the Qt using for building is deployed inside the AppImage. So it does matter, believe me.

> In the first place I have released many versions of QuickViewer in this environment, and it is no longer the first time I succeed.

Even if you succeed to release QuickViewer using this environment without `linuxdeployqt` that does not guarantee that `linuxdeployqt` can work with it without issues. Hence I asked to try a build using the exact OS and Qt combination from the project's `README.md` Travis CI example to see if the issue persists.

**Comment by kanryu** [comment]
When I write the conclusion earlier, I could release it.

Just deleting the build directory of QuickViewer with QtCreator did not finish, and it was necessary to delete the QVproject.pro.user file.

Since linuxdeployqt did not work properly in the continous build version, it has returned to the previous release(May 7 2017)

The difference between successful and unsuccessful deploy remains unknown.

I tried to submit AppDir where deply succeeded but I uploaded it because I exceed 10MB.
I will upload you elsewhere if necessary.
